src/runtime/hexagon_dma_device_shim.h

```diff
@@ -0,0 +1,207 @@
+/*!
+ * This file has the necessary structures and APIs for Initiating, executing and finishing a Hexagon DMA transfer
+ * The functions in this header file are the interfacing functions between Halide runtime and the Hexagon DMA driver
+ * The functions in this file lead to the hexagon DMA driver calls in case of availability of DMA driver and hexagon DMA tools
+ * In case of un availabilty of hexagon SDK tools and DMA drivers these function while mimic DMA with dummy transfers
+ */
+
+#ifndef _DMA_DEVICE_SHIM_H_
+#define _DMA_DEVICE_SHIM_H_
+
+#include "mini_qurt.h"
+#include "hexagon_mini_dma.h"
+using namespace Halide::Runtime::Internal::Qurt;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+__inline static int ceiling(int num, int div) {
+    return ((num + div -1)/(div));
+}
+
+__inline static int align(int x,int a) {
+    return (ceiling(x, a) * a);
```

Comment:
Not sure if this is used in an inner loop or not, but it uses integer division, which is very slow on Hexagon. There are faster ways to do alignment, if you can assume a is a power of 2, and maybe if x is unsigned.